analyzer/src/namespace/scopes.rs

```diff
@@ -101,8 +101,8 @@ impl ModuleScope {
     }
 
     /// Gets a type definition by name.
-    pub fn get_type_def(&self, name: &str) -> Option<Type> {
```

Comment:
This was just changed for consistency in the PR but the alternative that I proposed in the PR description would actually be exactly this, taking`&str` everywhere. Maybe that would be better after all?

Reply:
I think this is the way we should be doing it. What you already have is an improvement, but what you're suggesting here is probably the most rusty approach.

Reply:
Agreed! Will open up a new PR